libnavui-base/src/main/java/com/mapbox/navigation/ui/base/model/speedlimit/SpeedLimitFormatter.kt

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.navigation.ui.base.model.speedlimit
+
+import android.content.Context
+import com.mapbox.navigation.base.speed.model.SpeedLimitSign
+import com.mapbox.navigation.base.speed.model.SpeedLimitUnit
+import com.mapbox.navigation.ui.base.R
+import com.mapbox.navigation.ui.base.formatter.ValueFormatter
+import com.mapbox.navigation.ui.base.internal.speedlimit.SpeedLimitViewConstants
+import kotlin.math.roundToInt
+
+/**
+ * Formats speed limit data to a string for displaying to a user.
+ */
+class SpeedLimitFormatter(
+    context: Context
+) : ValueFormatter<SpeedLimitState.UpdateSpeedLimit, String> {
+
+    private val appContext = context.applicationContext
+
+    /**
+     * Formats speed limit data.
+     *
+     * @param state a state containing speed limit related data
+     *
+     * @return a formatted string
+     */
+    override fun format(state: SpeedLimitState.UpdateSpeedLimit): String {
+        return getSpeedLimit(state.getSignFormat(), state.getSpeedUnit(), state.getSpeed())
+    }
+
+    private fun getSpeedLimit(
+        sign: SpeedLimitSign,
+        unit: SpeedLimitUnit,
+        speedLimitKmph: Int
+    ): String {
+        return if (sign === SpeedLimitSign.VIENNA) {
+            if (unit === SpeedLimitUnit.KILOMETRES_PER_HOUR) {
+                appContext.getString(R.string.max_speed, speedLimitKmph)
+            } else {
+                val speed = (
+                    5 * (speedLimitKmph * SpeedLimitViewConstants.KILO_MILES_FACTOR / 5)
```

Comment:
Do we need to explicitly handle an infinite speed limit, e.g., on the Autobahn?

Reply:
I don't know what value comes from NN on that road.

Reply:
cc @mapbox/navnative